vector/src/main/java/im/vector/app/AppStateHandler.kt

```diff
@@ -59,7 +59,21 @@ class AppStateHandler @Inject constructor(
 
     val selectedRoomGroupingObservable = selectedSpaceDataSource.observe()
 
-    fun getCurrentRoomGroupingMethod(): RoomGroupingMethod? = selectedSpaceDataSource.currentValue?.orNull()
+    fun getCurrentRoomGroupingMethod(): RoomGroupingMethod? {
+        // XXX we should somehow make it live :/ just a work around
+        // For example just after creating a space and switching to it the
+        // name in the app Bar could show Empty Room, and it will not update unless you
+        // switch space
+        return selectedSpaceDataSource.currentValue?.orNull()?.let {
```

Comment:
I don't get it, why it's not lived? its an observable datasource

Reply:
The current space selection is live, but not the space details themselves (e.g if you rename the space, the observable will not notify of the change).
Making it more live is making the drawer lagging